Engineering/CodeReviews/TypeScript.md

```diff
@@ -0,0 +1,80 @@
+# TypeScript Code Reviews
+
+## Style Guide
+
+We use [`Prettier`](https://prettier.io/) to do code formatting for TypeScript. Using an automated code formatting tool like Prettier enforces a well accepted style guide that was collaboratively built by a wide range of companies including Microsoft, Facebook, and AirBnB. It is highly recommended that you use one of the native editor integrations and enable format on save. In addition, by default, we use the following overrides (shown here in the config format for VSCode) to standardize on single quotes and a 4-space indent:
+
+```json
+{
+    ...
+    "prettier.singleQuote": true,
+    "prettier.tabWidth": 4
+}
+```
+
+For higher level style guidance not covered by prettier, we follow the [AirBnB Style Guide](https://github.com/airbnb/javascript).
+
+## Linter
+
+All developers should run `TSLint` in a pre-commit hook to ensure standard formatting. We highly recommend using an editor integration like [vscode-tslint](https://github.com/Microsoft/vscode-tslint) to provide realtime feedback.
```

Comment:
Recommended: `lint-staged` and `husky`: https://github.com/okonet/lint-staged#installation-and-setup

- `husky` is a very popular tool for running pre-commit hooks and preventing commits whose hooks fail
- `lint-staged` ensures that only staged files get checked to determine if the pre-commit hooks need to be run